src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -3965,10 +3979,25 @@ void c_typecheck_baset::disallow_history_variables(const exprt &expr) const
   }
 
   if(expr.id() == ID_old)
+  {
+    error().source_location = expr.source_location();
+    error() << CPROVER_PREFIX "old expressions are not allowed" << eom;
```

Comment:
Are not allowed where? Could you improve this error message?

Reply:
Since it's a common error message for disallowing __CPROVER_old` within requires clauses and loop invariants, we shouldn't say "in __CPROVER_requires clauses". May be we could say:

```
 error: __CPROVER_old expressions are not allowed in this context
```